### PR TITLE
txn: add rollback log for pipelined dml

### PR DESCRIPTION
**Pull request by you06**
### What is changed and how it works?

Issue Number: ref #16291

What's Changed:

```commit-message
Add rollback log for pipelined dml for better diagnosis.
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

The log be like

```log
[check_txn_status.rs:180] ["flushed lock has been rollbacked"] [current_ts=18446744073709551615] [lock="Lock { lock_type: Put, primary_key: 7480000000000002BF5F698000000000000001037FFFFFFFFF676982038000000000000002, start_ts: TimeStamp(449257004056182793), ttl: 1361000, short_value: 30, for_update_ts: TimeStamp(0), txn_size: 18446744073709551615, min_commit_ts: TimeStamp(449257004056182794), use_async_commit: false, secondaries: [], rollback_ts: [], last_change: Unknown, txn_source: 0, is_locked_with_conflict: false, generation: 1 }"] [thread_id=64]
```

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: src/storage/txn/actions/check_txn_status.rs [hunk 1]
```diff
@@ -176,6 +176,12 @@ pub fn check_txn_status_lock_exists(
         }
         assert!(check_result.0.is_none() && check_result.1.is_none());
     } else if lock.ts.physical() + lock.ttl < current_ts.physical() {
+        if lock.generation > 0 {
+            warn!("flushed lock has been rollbacked";
```
Comment (ekexium) [inline src/storage/txn/actions/check_txn_status.rs:180]: Let's also log caller_start_ts to show who calls the rollback.

Comment (ekexium) [inline src/storage/txn/actions/check_txn_status.rs:180]:
```suggestion
            warn!("flushed lock has been rolled back";
```

**Review by cfzjywxk — APPROVED** [review]

**Review by ekexium — APPROVED** [review]

**Pull request merged**